include/libcyphal/application/node/heartbeat.hpp

```diff
@@ -0,0 +1,198 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_APPLICATION_NODE_HEARTBEAT_HPP_INCLUDED
+#define LIBCYPHAL_APPLICATION_NODE_HEARTBEAT_HPP_INCLUDED
+
+#include "libcyphal/executor.hpp"
+#include "libcyphal/presentation/presentation.hpp"
+#include "libcyphal/presentation/publisher.hpp"
+#include "libcyphal/transport/transport.hpp"
+#include "libcyphal/types.hpp"
+
+#include <cetl/pf17/cetlpf.hpp>
+#include <cetl/pmr/function.hpp>
+
+#include <uavcan/node/Heartbeat_1_0.hpp>
+
+#include <chrono>
+#include <cstddef>
+#include <cstdint>
+#include <utility>
+
+namespace libcyphal
+{
+namespace application
+{
+namespace node
+{
+
+/// @brief Defines 'Heartbeat' component for the application node.
+///
+/// No Sonar cpp:S4963 'The "Rule-of-Zero" should be followed'
+/// b/c we do directly handle resources here (namely capturing of `this` in the periodic callback).
+///
+class Heartbeat final  // NOSONAR cpp:S4963
```

Comment:
Should the names be more explicit? Perhaps `HeartbeatPublisher`

Reply:
I will go with `Producer` and `Provider` suffixes for Heartbeat and GetInfo entities.